core/commands/IssueViewPageCommand.php

```diff
@@ -235,10 +235,11 @@ protected function process() {
 			access_has_bug_level( config_get( 'update_bug_assign_threshold', config_get( 'update_bug_threshold' ) ), $t_issue_id );
 		$t_flags['can_change_status'] = !$t_issue_readonly && access_has_bug_level( config_get( 'update_bug_status_threshold' ), $t_issue_id );
 
-		$t_flags['can_monitor'] = !$t_force_readonly && !$t_anonymous_user &&
+		$t_monitor_flag = !$t_force_readonly && !$t_anonymous_user
+			&& user_is_monitoring_bug( auth_get_current_user_id(), $t_issue_id );
+		$t_flags['can_monitor'] = $t_monitor_flag &&
```

Comment:
Can't work.
$t_monitor_flag is just true if user_is_monitoring_bug
So you will get the monitor button just if you are already monitoring.

Reply:
Duh... you're right I don't know what I was thinking ! Will fix.